common/prque/prque_byteSlice_test.go

```diff
@@ -0,0 +1,92 @@
+// CookieJar - A contestant's algorithm toolbox
+// Copyright (c) 2013 Peter Szilagyi. All rights reserved.
+//
+// CookieJar is dual licensed: use of this source code is governed by a BSD
+// license that can be found in the LICENSE file. Alternatively, the CookieJar
+// toolbox may be used in accordance with the terms and conditions contained
+// in a signed written agreement between you and the author(s).
+
+package prque
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/klaytn/klaytn/common/math"
+
+	"github.com/klaytn/klaytn/common"
+)
+
+func TestPrqueByteSlice(t *testing.T) {
+	// Generate a batch of random data and a specific priority order
+	size := 16 * blockSize
+	data := make([]int, size)
+	prio := make([][]byte, size)
+	for i := 0; i < size; i++ {
+		data[i] = rand.Int()
+		prio[i] = common.MakeRandomBytes(256)
+	}
+	queue := NewByteSlice()
+	for rep := 0; rep < 2; rep++ {
+		// Fill a priority queue with the above data
+		for i := 0; i < size; i++ {
+			queue.Push(data[i], prio[i])
+			if queue.Size() != i+1 {
+				t.Errorf("queue size mismatch: have %v, want %v.", queue.Size(), i+1)
+			}
+		}
+		// Create a map the values to the priorities for easier verification
+		dict := make(map[uint64]int)
+		for i := 0; i < size; i++ {
+			dict[common.ByteBigEndianToInt64(prio[i])] = data[i]
+		}
+		// Pop out the elements in priority order and verify them
+		prevPrio := uint64(math.MaxUint64)
+		for !queue.Empty() {
+			val, prio := queue.Pop()
+			if common.ByteBigEndianToInt64(prio) > prevPrio {
+				t.Errorf("invalid priority order: %v after %v.", prio, prevPrio)
+			}
+			prevPrio = common.ByteBigEndianToInt64(prio)
+			if val != dict[prevPrio] {
+				t.Errorf("push/pop mismatch: have %v, want %v.", val, dict[prevPrio])
+			}
+			delete(dict, prevPrio)
+		}
```

Comment:
How about ensuring `len(dict) == 0` when the loop is finished?

Reply:
review reflected